# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSContext.cpp
# Commit: 9364941c6d53
# Full Hash: 9364941c6d538953131262f5d2c78163df803b3d
# Author: Boris Zbarsky <bzbarsky@mit.edu>
# Date: 2020-02-27 03:39:37
# Description:
#   Bug 1617527.  Remove the same-compartment checks when setting an exception on a JSContext r=sfink
#   
#   The exception is not guaranteed to be in the context compartment anyway (e.g. changing the compartment does not wrap the exception), so there isn't much point checking it up front.  And the exception stack is always stored as an unwrapped object, so again there's no point checking it on set, since we just plan to UncheckedUnwrap it.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D64004
# ==============================================================================

diff -r 5fc409bc3530 -r 9364941c6d53 js/src/vm/JSContext.cpp
--- a/js/src/vm/JSContext.cpp	Wed Feb 26 23:39:02 2020 +0200
+++ b/js/src/vm/JSContext.cpp	Wed Feb 26 21:17:15 2020 +0000
@@ -1445,7 +1445,6 @@
   this->throwing = true;
   this->unwrappedException() = v;
   this->unwrappedExceptionStack() = stack;
-  check(v);
 }
 
 void JSContext::setPendingExceptionAndCaptureStack(HandleValue value) {